kernel/arch/dreamcast/kernel/rtc.c

```diff
@@ -22,13 +25,20 @@
  */
 
 #include <arch/rtc.h>
+#include <arch/timer.h>
 #include <dc/g2bus.h>
 
+#define RTC_CONTROL_ADDR        0xa0710008  /* SFR with write-enable bit */
+#define RTC_SECONDS_HIGH_ADDR   0xa0710000  /* Upper 16 bits of timestamp */
+#define RTC_SECONDS_LOW_ADDR    0xa0710004  /* Lower 16 bits of timestamp */
```

Comment:
I'd suggest that the three register values be in address order (CTRL last), and be moved into the header file. With them in the header they can get doxy comments added, which would help tracking and hw documentation. Then could also note the behavior that is in comments of the set time function with the CTRL being write-only and self clearing when the high_addr is written to.

Though it's possibly overly pedantic, I might also have '#define RTC_CTRL_ADDR_WRITE_ENABLE (1 << 0)' or similar.
Or having RTC_CONTROL_ADDR be RTC_WRITE_ENABLE then define the value as RTC_WRITE_ENABLE_SET. In any case tagging in #151 for visibility there.

Reply:
I had actually considered moving the registers to the header file, but was worried you guys might think that was beyond the scope of what this driver should be doing...

BUT I ACTUALLY TOTALLY AGREE! That's a fantastic idea. I think every header should document the hardware publicly as well, like that for our KOS hardware documentation in doxygen!

Agreed on what you're saying about having a define for the actual write enable bit. I'll get on this.